geo/src/main/java/org/odk/collect/geo/selection/SelectionMapFragment.kt

```diff
@@ -376,7 +387,7 @@ class SelectionMapFragment(
             ids + map.addPoly(item.points, false, false)
         }
 
-        items.zip(pointIds + traceIds).forEach { (item, featureId) ->
+        (singlePoints + traces).zip(pointIds + traceIds).forEach { (item, featureId) ->
```

Comment:
Before, these were being zipped in an order that just happened to work if points and traces were defined in the data as a block of points and then a block of traces.